Mixpanel/MPSwizzler.m

```diff
@@ -128,50 +128,54 @@ + (BOOL)isLocallyDefinedMethod:(Method)aMethod onClass:(Class)aClass
 
 + (void)swizzleSelector:(SEL)aSelector onClass:(Class)aClass withBlock:(swizzleBlock)aBlock named:(NSString *)aName
 {
-    Method aMethod = class_getInstanceMethod(aClass, aSelector);
-    if (aMethod) {
-        uint numArgs = method_getNumberOfArguments(aMethod);
-        if (numArgs >= MIN_ARGS && numArgs <= MAX_ARGS) {
-
-            BOOL isLocal = [self isLocallyDefinedMethod:aMethod onClass:aClass];
-            IMP swizzledMethod = (IMP)mp_swizzledMethods[numArgs - 2];
-            MPSwizzle *swizzle = [self swizzleForMethod:aMethod];
-
-            if (isLocal) {
-                if (!swizzle) {
-                    IMP originalMethod = method_getImplementation(aMethod);
-
-                    // Replace the local implementation of this method with the swizzled one
-                    method_setImplementation(aMethod,swizzledMethod);
-
-                    // Create and add the swizzle
-                    swizzle = [[MPSwizzle alloc] initWithBlock:aBlock named:aName forClass:aClass selector:aSelector originalMethod:originalMethod withNumArgs:numArgs];
-                    [self setSwizzle:swizzle forMethod:aMethod];
-
+    @try {
```

Comment:
The intention of my first comment was to make you consider changing the `try/catch` because we should not raise exceptions. When I asked about the multiple places it could crash, it was so you see that we are, again, raising exceptions in those places. 
I'll wait for a modified version with proper exception handling.